### PR TITLE
Feature form's text widget implementation

### PR DESCRIPTION
**Pull request by nirvn**
This PR implements support for the {attribute,feature} form's text widget. The implementation takes a slightly different path from that of the QML and HTML container and apply the new path to the latter two; I'm hoping will result in faster feature form loading as it kills a few unnecessary QML item creations.

The PR also makes the text editor widget (as well as its QML and HTML siblings) dynamic and responsive to feature attribute value change as the feature is being edited.

Simple example: notice how the text editor's "parent feature details" text value changes as I'm picking a different parent feature in a parent<->child relationship:

[Screencast from 2023-08-23 18-22-42.webm](https://github.com/opengisch/QField/assets/1728657/5326beac-3aa3-4491-bcbe-73db155cdf7d)

### CODE REVIEW

**Comment by nirvn** [comment]
@ZsanettMed , as agreed in yesterday's workshop.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for faccd3b17f018e610c966bc00c65ffaecf470535: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4528-02e40280dc7b77a0b68a27999b9d52174abb11b7-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4528-02e40280dc7b77a0b68a27999b9d52174abb11b7-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4528-02e40280dc7b77a0b68a27999b9d52174abb11b7-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4528-02e40280dc7b77a0b68a27999b9d52174abb11b7-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4528-02e40280dc7b77a0b68a27999b9d52174abb11b7-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4528-02e40280dc7b77a0b68a27999b9d52174abb11b7-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4528-02e40280dc7b77a0b68a27999b9d52174abb11b7-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4528-02e40280dc7b77a0b68a27999b9d52174abb11b7-x86-android.apk)
</details>

File: src/core/attributeformmodelbase.cpp [hunk 1]
```diff
@@ -597,6 +622,97 @@ void AttributeFormModelBase::updateDefaultValues( int fieldIndex, QVector<int> u
...
+  QMap<QStandardItem *, QString>::ConstIterator editorWidgetCodesIterator( mEditorWidgetCodes.constBegin() );
+  for ( ; editorWidgetCodesIterator != mEditorWidgetCodes.constEnd(); editorWidgetCodesIterator++ )
+  {
+    QStandardItem *item = editorWidgetCodesIterator.key();
+    QString code = editorWidgetCodesIterator.value();
+    bool needUpdate = false;
+
+    if ( item->data( AttributeFormModel::ElementType ) == QStringLiteral( "qml" ) || item->data( AttributeFormModel::ElementType ) == QStringLiteral( "html" ) )
+    {
+      const thread_local QRegularExpression sRegEx( "expression\\.evaluate\\(\\s*\\\"(.*?[^\\\\])\\\"\\s*\\)", QRegularExpression::MultilineOption | QRegularExpression::DotMatchesEverythingOption );
```
Comment (suricactus) [inline src/core/attributeformmodelbase.cpp:640]: /// :exploding_head: ///

**Review by suricactus — APPROVED** [review]
Followed most of the logic, looks good.

Code also looks good. No idea if we can do something about the regex that is terrible to read due to escaping.

Also tested on mobile, looks great.

Sidenote - the Text Widget seems undocumented to me, so it would be great to update both QGIS and QField docs pointing it's perfect support.

**Comment by nirvn** [comment]
@suricactus , thanks for the review. Documentation gap is indeed an issue.

**Pull request merged**